sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAzureNamedKeyCredentialPolicy.java

```diff
@@ -74,37 +78,34 @@ String generateAuthorizationHeader(URL requestUrl, Map<String, String> headers)
      *
      * @return A string to sign for the request.
      */
-    private String buildStringToSign(URL requestUrl, Map<String, String> headers) {
-        String dateHeader = headers.containsKey("x-ms-date")
+    private String buildStringToSign(URL requestUrl, HttpHeaders headers) {
+        // If the x-ms-header exists ignore the Date header.
+        String dateHeader = headers.get("x-ms-date") != null
             ? ""
             : this.getStandardHeaderValue(headers, "Date");
 
-        String s = String.join("\n",
-            dateHeader,  //date
-            getCanonicalizedResource(requestUrl)); //Canonicalized resource
-
-        return s;
+        return String.join("\n", dateHeader, getCanonicalizedResource(requestUrl));
```

Comment:
Can `getCanonicalizedResource` be made static so this method can be made static?

Reply:
Yes, we can do that.